src/main/java/com/thalesgroup/hudson/plugins/sourcemonitor/FileStats.java

```diff
@@ -0,0 +1,174 @@
+    }
+
+    public int getNumStatements() {
+        return numStatements;
+    }
+
+    public void setNumStatements(int numStatements) {
+        this.numStatements = numStatements;
+    }
+
+    public int getMaxComplexity() {
+        return maxComplexity;
+    }
+
+    public void setMaxComplexity(int maxComplexity) {
+        this.maxComplexity = maxComplexity;
+    }
+
+    public void setParameters(ConfigurableParameters parameters) {
+        this.parameters = parameters;
+    }
+
+    public void setParentFile(File parentFile) {
+        this.parentFile = parentFile;
+    }
+
+    /** HTML Generation */
+    public String getFunctionOutput(){
+        StringBuilder builder = new StringBuilder();
+        for (int i=0; i<functionStats.size();i++){
+            builder.append("<tr><td>");
+            builder.append(functionStats.get(i).getFunction());
+            builder.append("</td>");
+            String str1;
+
+            int numState = functionStats.get(i).getStatements();
+            int paramHealth = SourceMonitorUtility.getStatementsHealth(parameters, numState);
+            builder.append(SourceMonitorUtility.getColoredString(paramHealth, numState));
+            builder.append("</td>");
+
+            int maxComp = functionStats.get(i).getComplexity();
+            paramHealth = SourceMonitorUtility.getComplexityHealth(parameters, maxComp);
+            builder.append(SourceMonitorUtility.getColoredString(paramHealth, maxComp));
+            builder.append("</td>");
+            builder.append("</td></tr>");
+        }
+        return builder.toString();
+    }
+
+    public String createCodeTable(File source) throws IOException, InterruptedException {
+
+        FileInputStream inputStream = null;
+        InputStreamReader isReader = null;
+        BufferedReader input = null;
+        StringBuilder builder = new StringBuilder();
+        int line = 0;
+
+        try {
+            inputStream = new FileInputStream(source);
+            isReader = new InputStreamReader(inputStream, "UTF-8");
+            input = new BufferedReader(isReader);
+            String content;
+            while ((content = input.readLine()) != null) {
+                line++;
+
+                builder.append("<tr style = \"background-color:white\" class=\"noCover\">\n");
+                builder.append("<td style=\"padding-left:2px;text-align:right\"><a name='" + line + "'/>" + line + "</td>\n");
+                builder.append("<td style=\"padding-left:6px;padding-right:4px\">"
+                        + content.replace("&", "&amp;").replace("<", "&lt;").replace(">", "&gt;").replace("\n", "").replace("\r", "").replace(" ",
+                        "&nbsp;").replace("\t", "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;") + "</td>\n");
+                builder.append("</tr>\n");
+            }
+            input.close();
+            inputStream.close();
+            isReader.close();
+        } catch(Exception e) {
```

Comment:
Shouldn't we use a try with resources or at least a try...finally to properly close our resources on error.